pkg/secrethub/acl.go

```diff
@@ -286,3 +292,93 @@ func (c *Client) getAccessLevel(path api.BlindNamePath, accountName api.AccountN
 
 	return accessLevel, nil
 }
+
+// Iterator returns an iterator that retrieves all access rules that apply to a
+// directory.
+func (s accessRuleService) Iterator(path string, params *AccessRuleIteratorParams) AccessRuleIterator {
+	var depth int
+	var ancestors bool
+	if params == nil {
+		depth = -1
+		ancestors = false
+	} else {
+		depth = params.depth
+		ancestors = params.ancestors
+	}
+
+	data, err := s.List(path, depth, ancestors)
+	return &accessRuleIterator{
+		index: 0,
+		data:  data,
+		err:   err,
+	}
+}
+
+// AccessRuleIteratorParams specify parameters used when listing access rules.
+type AccessRuleIteratorParams struct {
+	depth     int  // depth defines the depth of traversal for the iterator, -1 means listing all subdirectories.
```

Comment:
`int` has a `zero-value` of 0. As the zero value will be used when constructing the params struct without specifying the field, the zero-value will act as the default value.

For this function, I think returning access rules for all subdirectories (currently specified with `-1`) would be a better default for this function.

To be able to use a value other than zero as the default value, we could change the type to `*int` and use the default value when the value isn't explicitly set (when the value is the zero-value of a pointer value: `nil`).

If we want all subdirectories to be the default, we can even change the type to `*uint`, so that negative values are not allowed.

Reply:
Good point! I'll use `*uint`.